client/src/modules/shopping-list/components/InputBar/InputBar.jsx

```diff
@@ -69,17 +76,20 @@ class InputBar extends Component {
 
 InputBar.propTypes = {
   currentUser: UserPropType.isRequired,
+  match: PropTypes.objectOf(PropTypes.any),
```

Comment:
You can declare match prop with details and specify all needed params types:  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
...
    })
  }).isRequired,